mplc/experiment.py

```diff
@@ -0,0 +1,143 @@
+# -*- coding: utf-8 -*-
+"""
+An Experiment regroups multiple scenarios and enables to run them and analyze their results.
+"""
+
+import datetime
+import os
+from pathlib import Path
+
+import matplotlib.pyplot as plt
+from loguru import logger
+
+from . import scenario as scenario_module
+from . import utils, constants
+
+DEFAULT_CONFIG_FILE = "../config.yml"
+
+
+class Experiment:
+    def __init__(
+            self,
+            experiment_name=None,
+            nb_repeats=1,
+            scenarios_list=[]
+    ):
+        """
+        :param experiment_name: string, name of the experiment
+        :param nb_repeats: int, number of repeats of the experiments (as an experiment includes
+                           a number of non-deterministic phenomena. Example: 5
+        :param scenarios_list: list, list of scenarios to be run during the experiment. scenario can also be added via
+        """
+
+        self.name = experiment_name
+        if experiment_name:
+            self.experiment_path = self.define_experiment_path()
+        self.scenarios_list = []
+        for scenario in scenarios_list:
+            self.add_scenario(scenario)
+        self.nb_repeats = nb_repeats
+
+    def define_experiment_path(self):
+        """Define the path and create folder for saving results of the experiment"""
+
+        now = datetime.datetime.now()
+        now_str = now.strftime("%Y-%m-%d_%Hh%M")
+
+        full_experiment_name = self.name + "_" + now_str
+        experiment_path = Path.cwd() / constants.EXPERIMENTS_FOLDER_NAME / full_experiment_name
+
+        # Check if experiment folder already exists
+        while experiment_path.exists():
```

Comment:
Just wondering, why not a simple `if` check?

Reply:
Because we loop to add `_bis` to the experiment name until the name does not already exist
However, That will be changed. We now use the current data/time